packages/extras/src/lib/interactivity/setupInteractivity.ts

```diff
@@ -61,60 +58,46 @@ export const setupInteractivity = (state: State) => {
     }
   }
 
-  const enabled = memoize(state.enabled)
+  const enabled = memoize(context.enabled)
 
   const getHits = (): Intersection[] => {
-    const duplicates = new Set<string>()
+    if (!enabled) return []
 
     const intersections: Intersection[] = []
-
-    let hits = state.interactiveObjects
-      .flatMap((obj) => (enabled.current ? state.raycaster.intersectObject(obj, true) : []))
-      // Sort by distance
-      .sort((a, b) => a.distance - b.distance)
-      // Filter out duplicates
-      .filter((item) => {
-        const id = getIntersectionId(item as Intersection)
-        if (duplicates.has(id)) return false
-        duplicates.add(id)
-        return true
-      })
-
-    if (state.filter) hits = state.filter(hits, state)
+    const hits = context.raycaster.intersectObjects(context.interactiveObjects, true)
```

Comment:
The `intersectObjects` method sorts by distance and contains no duplicates as part of its behavior, allowing us to simplify `getHits()`

Reply:
[can confirm](https://threejs.org/docs/index.html?q=ray#api/en/core/Raycaster.intersectObjects), [source link too](https://github.com/mrdoob/three.js/blob/841d2e791d3e8a2463322c5ca31b16956828b91c/src/core/Raycaster.js#L66)